### ISSUE TITLE
system.HasFocus returning true when minimized with task-bar while in Windowed-mode

### ISSUE THREAD

**Issue by yupi2** [Facepunch/garrysmod-issues#2424]
Windows 7 with non-Dev branch.

```
] version
Protocol version 24
Exe version 16.01.01 (garrysmod)
Exe build: 09:39:42 Jan 13 2016 (6276) (4000)
```

Reproduction steps:
- Start single-player server.
- Open console.
- `sv_cheats 1`
- `lua_run_cl timer.Simple(4, function() chat.AddText(tostring(system.HasFocus())) end)`
- Close console so game resumes & open the chat-box to free the mouse cursor.
- Click on the GMOD icon in the task-bar to minimize GMOD.
- Wait a moment.
- Open GMOD's window and see "true" in chat.

It's obviously not a game-breaking bug, but it might as well be reported.

**Comment by WinterPhoenix** [comment]
This doesn't seem to happen anymore, or doesn't happen on Windows 10.
```
] version
Protocol version 24
Exe version 2020.03.17 (garrysmod)
Exe build: 19:20:26 Jun 10 2020 (7881) (4000)
```

Can anybody else confirm?

**Comment by robotboy655** [comment]
It definitely happens for me on Win10, but only if I actually click on the game in the taskbar, it doesn't happen if I minimize it from top right corner.

**Comment by robotboy655** [comment]
It's an issue with WM_ACTIVATEAPP being called back with "activating" state immediately after being called with the "deactivating" state, so this is more of a Windows issue. I have tested this with a 3rd party application, and the same issue persists, so not it's not just GMod.

**Comment by WinterPhoenix** [comment]
On Windows 10 1909 at least, it looks like you're right. Interesting bug.

**Comment by viral32111** [comment]
> Still would like some confirmations from other people, one way or the other.

I've just tested the steps in the original post and I can confirm it does happen. I'm on Windows 10 20H2 (Build [19042.330](https://blogs.windows.com/windowsexperience/2020/06/16/announcing-windows-10-insider-preview-build-19042-330-20h2/)) using Garry's Mod 2020.06.26 (Main branch) (`-noaddons` & `-noworkshop`). I got the exact same results as you did, `true` when minimised by clicking the game on the taskbar, but `false` when minimised using actual minimise button on the window.

![aa](https://user-images.githubusercontent.com/19510403/86913050-91054a00-c10d-11ea-803f-bd9f8d54538d.png)
